Camel-K: Listing all kamelets fails with UnrecognizedPropertyException

### Describe the bug

camel-k-client should be aligned with 1.7.0 camel-k, but listing the default kamelets installed by 1.7.0 operator fails with UnrecognizedPropertyException

for example `avro-deserialize-action` kamelet contains `spec.definition.properties.validate.x-descriptors` that is not recognized:

```
oc get kamelet avro-deserialize-action -o jsonpath='{.spec.definition.properties.validate}'
```

This issue can be worked around by calling `        Serialization.jsonMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);` before calling the `kamelets().list()` method

### Fabric8 Kubernetes Client version

other (please specify in additional context)

### Steps to reproduce

1. install camel-k 1.7.0, for example with `kamel` binary 1.7.0 version `kamel install --olm=false`
2. execute `new DefaultCamelKClient().v1alpha1().kamelets().list();`

### Expected behavior

list all kamelets

### Runtime

OpenShift

### Kubernetes API Server version

1.22.3@latest

### Environment

Linux

### Fabric8 Kubernetes Client Logs

```shell
io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
    at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable (KubernetesClientException.java:103)
    at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable (KubernetesClientException.java:97)
    at io.fabric8.kubernetes.client.utils.Serialization.unmarshal (Serialization.java:319)
    at io.fabric8.kubernetes.client.utils.Serialization.unmarshal (Serialization.java:271)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse (OperationSupport.java:557)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse (OperationSupport.java:518)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse (OperationSupport.java:502)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.listRequestHelper (BaseOperation.java:133)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list (BaseOperation.java:415)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list (BaseOperation.java:404)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list (BaseOperation.java:83)
    at org.example.Main.main (Main.java:10)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:254)
    at java.lang.Thread.run (Thread.java:829)
Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "x-descriptors" (class io.fabric8.kubernetes.api.model.apiextensions.v1.JSONSchemaProps), not marked as ignorable (44 known properties: "definitions", "default", "x-kubernetes-list-map-keys", "$schema", "multipleOf", "x-kubernetes-validations", "x-kubernetes-int-or-string", "minItems", "exclusiveMinimum", "not", "allOf", "additionalItems", "id", "oneOf", "required", "maximum", "nullable", "exclusiveMaximum", "anyOf", "minProperties", "externalDocs", "dependencies", "example", "x-kubernetes-preserve-unknown-fields", "$ref", "properties", "format", "additionalProperties", "description", "enum", "minimum", "pattern", "minLength", "x-kubernetes-map-type", "x-kubernetes-embedded-resource", "patternProperties", "maxLength", "uniqueItems", "maxProperties", "type", "maxItems", "items", "title", "x-kubernetes-list-type"])
 at [Source: (BufferedInputStream); line: 1, column: 6080] (through reference chain: io.fabric8.camelk.v1alpha1.KameletList["items"]->java.util.ArrayList[0]->io.fabric8.camelk.v1alpha1.Kamelet["spec"]->io.fabric8.camelk.v1alpha1.KameletSpec["definition"]->io.fabric8.kubernetes.api.model.apiextensions.v1.JSONSchemaProps["properties"]->java.util.LinkedHashMap["validate"]->io.fabric8.kubernetes.api.model.apiextensions.v1.JSONSchemaProps["x-descriptors"])
    at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from (UnrecognizedPropertyException.java:61)
    at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty (DeserializationContext.java:1127)
    at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty (StdDeserializer.java:1989)
    at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty (BeanDeserializerBase.java:1700)
    at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla (BeanDeserializerBase.java:1678)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize (BeanDeserializer.java:319)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize (BeanDeserializer.java:176)
    at com.fasterxml.jackson.databind.deser.std.MapDeserializer._readAndBindStringKeyMap (MapDeserializer.java:609)
    at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize (MapDeserializer.java:437)
    at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize (MapDeserializer.java:32)
    at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet (MethodProperty.java:129)
    at io.fabric8.kubernetes.client.utils.serialization.SettableBeanPropertyDelegate.deserializeAndSet (SettableBeanPropertyDelegate.java:131)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize (BeanDeserializer.java:313)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize (BeanDeserializer.java:176)
    at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet (MethodProperty.java:129)
    at io.fabric8.kubernetes.client.utils.serialization.SettableBeanPropertyDelegate.deserializeAndSet (SettableBeanPropertyDelegate.java:131)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize (BeanDeserializer.java:313)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize (BeanDeserializer.java:176)
    at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet (MethodProperty.java:129)
    at io.fabric8.kubernetes.client.utils.serialization.SettableBeanPropertyDelegate.deserializeAndSet (SettableBeanPropertyDelegate.java:131)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize (BeanDeserializer.java:313)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize (BeanDeserializer.java:176)
    at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer._deserializeFromArray (CollectionDeserializer.java:355)
    at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize (CollectionDeserializer.java:244)
    at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize (CollectionDeserializer.java:28)
    at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet (MethodProperty.java:129)
    at io.fabric8.kubernetes.client.utils.serialization.SettableBeanPropertyDelegate.deserializeAndSet (SettableBeanPropertyDelegate.java:131)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize (BeanDeserializer.java:313)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize (BeanDeserializer.java:176)
    at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue (DefaultDeserializationContext.java:322)
    at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose (ObjectMapper.java:4674)
    at com.fasterxml.jackson.databind.ObjectMapper.readValue (ObjectMapper.java:3674)
    at io.fabric8.kubernetes.client.utils.Serialization.unmarshal (Serialization.java:317)
    at io.fabric8.kubernetes.client.utils.Serialization.unmarshal (Serialization.java:271)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse (OperationSupport.java:557)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse (OperationSupport.java:518)
    at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse (OperationSupport.java:502)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.listRequestHelper (BaseOperation.java:133)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list (BaseOperation.java:415)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list (BaseOperation.java:404)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list (BaseOperation.java:83)
```


### Additional context

camel-k-client 5.12.1
openshift 4.9.17

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,7 @@
 
 #### Improvements
 * Remove `setIntVal`, `setStrVal`, `setKind` setters from `IntOrString` class to avoid invalid combinations
+* Fix #3852: Deserializing kamelets fails with UnrecognizedPropertyException
 * Fix #3889 : remove piped stream for file download
 * Fix #1285: removed references to manually calling registerCustomKind
 * Fix #3334: adding basic support for server side apply.  Use patch(PatchContext.of(PatchType.SERVER_SIDE_APPLY), service), or new PatchContext.Builder().withPatchType(PatchType.SERVER_SIDE_APPLY).withForce(true).build() to override conflicts.

--- a/extensions/camel-k/generator-v1alpha1/cmd/generate/generate.go
+++ b/extensions/camel-k/generator-v1alpha1/cmd/generate/generate.go
@@ -44,7 +44,6 @@ func main() {
 
 	// types that are manually defined in the model
 	providedTypes := []schemagen.ProvidedType{
-		{GoType: reflect.TypeOf(v1alpha1.JSONSchemaProps{}), JavaClass: "io.fabric8.kubernetes.api.model.apiextensions.v1.JSONSchemaProps"},
 		{GoType: reflect.TypeOf(v1.Flow{}), JavaClass: "com.fasterxml.jackson.databind.JsonNode"},
 		{GoType: reflect.TypeOf(v1.TraitConfiguration{}), JavaClass: "com.fasterxml.jackson.databind.JsonNode"},
 	}
@@ -76,11 +75,12 @@ func main() {
 
 	// overwriting some times
 	manualTypeMap := map[reflect.Type]string{
-		reflect.TypeOf(machinery.Time{}):              "java.lang.String",
-		reflect.TypeOf(apis.URL{}):                    "java.lang.String",
-		reflect.TypeOf(apis.VolatileTime{}):           "java.lang.String",
-		reflect.TypeOf(runtime.RawExtension{}):        "java.util.Map<String, Object>",
-		reflect.TypeOf(v1.Template{}):                 "java.util.Map<String, Object>",
+		reflect.TypeOf(machinery.Time{}):       "java.lang.String",
+		reflect.TypeOf(apis.URL{}):             "java.lang.String",
+		reflect.TypeOf(apis.VolatileTime{}):    "java.lang.String",
+		reflect.TypeOf(runtime.RawExtension{}): "java.util.Map<String, Object>",
+		reflect.TypeOf(v1.Template{}):          "java.util.Map<String, Object>",
+		reflect.TypeOf(v1alpha1.JSON{}):        "com.fasterxml.jackson.databind.JsonNode",
 		reflect.TypeOf(v1alpha1.EndpointProperties{}): "java.util.Map<String, Object>",
 	}
 

--- a/extensions/camel-k/model-v1alpha1/pom.xml
+++ b/extensions/camel-k/model-v1alpha1/pom.xml
@@ -132,9 +132,6 @@
                 <target>
                   <!-- removing the generated Schema class -->
                   <delete
-                    file="${project.build.directory}/generated-sources/io/fabric8/camelk/v1alpha1/JSONSchemaProps.java"
-                    verbose="true"/>
-                  <delete
                     file="${project.build.directory}/generated-sources/io/fabric8/camelk/api/model/CamelKSchemaV1alpha1.java"
                     verbose="true"/>
                 </target>

--- a/extensions/camel-k/model-v1alpha1/src/main/resources/schema/camel-k-schema-v1alpha1.json
+++ b/extensions/camel-k/model-v1alpha1/src/main/resources/schema/camel-k-schema-v1alpha1.json
@@ -46,10 +46,163 @@
           "type": "string"
         },
         "schema": {
-          "existingJavaType": "io.fabric8.kubernetes.api.model.apiextensions.v1.JSONSchemaProps"
+          "$ref": "#/definitions/github_com_apache_camel-k_pkg_apis_camel_v1alpha1_JSONSchemaProps",
+          "existingJavaType": "io.fabric8.camelk.v1alpha1.JSONSchemaProps"
         }
       },
       "javaType": "io.fabric8.camelk.v1alpha1.EventTypeSpec",
+      "javaInterfaces": [
+        "io.fabric8.kubernetes.api.model.KubernetesResource"
+      ]
+    },
+    "github_com_apache_camel-k_pkg_apis_camel_v1alpha1_ExternalDocumentation": {
+      "type": "object",
+      "properties": {
+        "description": {
+          "type": "string"
+        },
+        "url": {
+          "type": "string"
+        }
+      },
+      "javaType": "io.fabric8.camelk.v1alpha1.ExternalDocumentation",
+      "javaInterfaces": [
+        "io.fabric8.kubernetes.api.model.KubernetesResource"
+      ]
+    },
+    "github_com_apache_camel-k_pkg_apis_camel_v1alpha1_JSONSchemaProp": {
+      "type": "object",
+      "properties": {
+        "default": {
+          "existingJavaType": "com.fasterxml.jackson.databind.JsonNode"
+        },
+        "description": {
+          "type": "string"
+        },
+        "enum": {
+          "type": "array",
+          "javaOmitEmpty": true,
+          "items": {
+            "existingJavaType": "com.fasterxml.jackson.databind.JsonNode"
+          }
+        },
+        "example": {
+          "existingJavaType": "com.fasterxml.jackson.databind.JsonNode"
+        },
+        "exclusiveMaximum": {
+          "type": "boolean"
+        },
+        "exclusiveMinimum": {
+          "type": "boolean"
+        },
+        "format": {
+          "type": "string"
+        },
+        "id": {
+          "type": "string"
+        },
+        "maxItems": {
+          "type": "integer",
+          "existingJavaType": "Long"
+        },
+        "maxLength": {
+          "type": "integer",
+          "existingJavaType": "Long"
+        },
+        "maxProperties": {
+          "type": "integer",
+          "existingJavaType": "Long"
+        },
+        "maximum": {
+          "type": "string",
+          "existingJavaType": "String"
+        },
+        "minItems": {
+          "type": "integer",
+          "existingJavaType": "Long"
+        },
+        "minLength": {
+          "type": "integer",
+          "existingJavaType": "Long"
+        },
+        "minProperties": {
+          "type": "integer",
+          "existingJavaType": "Long"
+        },
+        "minimum": {
+          "type": "string",
+          "existingJavaType": "String"
+        },
+        "multipleOf": {
+          "type": "string",
+          "existingJavaType": "String"
+        },
+        "nullable": {
+          "type": "boolean"
+        },
+        "pattern": {
+          "type": "string"
+        },
+        "title": {
+          "type": "string"
+        },
+        "type": {
+          "type": "string"
+        },
+        "uniqueItems": {
+          "type": "boolean"
+        },
+        "x-descriptors": {
+          "type": "array",
+          "javaOmitEmpty": true,
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "javaType": "io.fabric8.camelk.v1alpha1.JSONSchemaProp",
+      "javaInterfaces": [
+        "io.fabric8.kubernetes.api.model.KubernetesResource"
+      ]
+    },
+    "github_com_apache_camel-k_pkg_apis_camel_v1alpha1_JSONSchemaProps": {
+      "type": "object",
+      "properties": {
+        "$schema": {
+          "type": "string"
+        },
+        "description": {
+          "type": "string"
+        },
+        "example": {
+          "existingJavaType": "com.fasterxml.jackson.databind.JsonNode"
+        },
+        "externalDocs": {
+          "$ref": "#/definitions/github_com_apache_camel-k_pkg_apis_camel_v1alpha1_ExternalDocumentation",
+          "existingJavaType": "io.fabric8.camelk.v1alpha1.ExternalDocumentation"
+        },
+        "id": {
+          "type": "string"
+        },
+        "properties": {
+          "type": "object",
+          "existingJavaType": "java.util.Map\u003cString,io.fabric8.camelk.v1alpha1.JSONSchemaProp\u003e"
+        },
+        "required": {
+          "type": "array",
+          "javaOmitEmpty": true,
+          "items": {
+            "type": "string"
+          }
+        },
+        "title": {
+          "type": "string"
+        },
+        "type": {
+          "type": "string"
+        }
+      },
+      "javaType": "io.fabric8.camelk.v1alpha1.JSONSchemaProps",
       "javaInterfaces": [
         "io.fabric8.kubernetes.api.model.KubernetesResource"
       ]
@@ -316,7 +469,8 @@
           "existingJavaType": "io.fabric8.camelk.v1alpha1.AuthorizationSpec"
         },
         "definition": {
-          "existingJavaType": "io.fabric8.kubernetes.api.model.apiextensions.v1.JSONSchemaProps"
+          "$ref": "#/definitions/github_com_apache_camel-k_pkg_apis_camel_v1alpha1_JSONSchemaProps",
+          "existingJavaType": "io.fabric8.camelk.v1alpha1.JSONSchemaProps"
         },
         "dependencies": {
           "type": "array",
@@ -394,6 +548,18 @@
     "github_com_apache_camel-k_pkg_apis_camel_v1alpha1_EventTypeSpec": {
       "$ref": "#/definitions/github_com_apache_camel-k_pkg_apis_camel_v1alpha1_EventTypeSpec",
       "existingJavaType": "io.fabric8.camelk.v1alpha1.EventTypeSpec"
+    },
+    "github_com_apache_camel-k_pkg_apis_camel_v1alpha1_ExternalDocumentation": {
+      "$ref": "#/definitions/github_com_apache_camel-k_pkg_apis_camel_v1alpha1_ExternalDocumentation",
+      "existingJavaType": "io.fabric8.camelk.v1alpha1.ExternalDocumentation"
+    },
+    "github_com_apache_camel-k_pkg_apis_camel_v1alpha1_JSONSchemaProp": {
+      "$ref": "#/definitions/github_com_apache_camel-k_pkg_apis_camel_v1alpha1_JSONSchemaProp",
+      "existingJavaType": "io.fabric8.camelk.v1alpha1.JSONSchemaProp"
+    },
+    "github_com_apache_camel-k_pkg_apis_camel_v1alpha1_JSONSchemaProps": {
+      "$ref": "#/definitions/github_com_apache_camel-k_pkg_apis_camel_v1alpha1_JSONSchemaProps",
+      "existingJavaType": "io.fabric8.camelk.v1alpha1.JSONSchemaProps"
     },
     "github_com_apache_camel-k_pkg_apis_camel_v1alpha1_Kamelet": {
       "$ref": "#/definitions/github_com_apache_camel-k_pkg_apis_camel_v1alpha1_Kamelet",

--- a/extensions/camel-k/model-v1alpha1/src/test/java/io/fabric8/camelk/KameletModelTest.java
+++ b/extensions/camel-k/model-v1alpha1/src/test/java/io/fabric8/camelk/KameletModelTest.java
@@ -17,12 +17,12 @@ package io.fabric8.camelk;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import org.junit.jupiter.api.Test;
-
+import io.fabric8.camelk.v1alpha1.JSONSchemaPropsBuilder;
 import io.fabric8.camelk.v1alpha1.Kamelet;
 import io.fabric8.camelk.v1alpha1.KameletBuilder;
-import io.fabric8.kubernetes.api.model.apiextensions.v1.JSONSchemaPropsBuilder;
+import org.junit.jupiter.api.Test;
 
+import java.util.Collections;
 import java.util.Scanner;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
@@ -33,17 +33,17 @@ import static org.junit.jupiter.api.Assertions.assertTrue;
 class KameletModelTest {
 
   @Test
-  void shouldCreateKamelet()  {
+  void shouldCreateKamelet() {
     Kamelet kamelet = new KameletBuilder()
-      .withNewMetadata()
-      .withName("my-kamelet")
-      .endMetadata()
-      .withNewSpec()
-      .withDefinition(new JSONSchemaPropsBuilder()
-                      // blah blah
-                      .build())
-      .endSpec()
-      .build();
+        .withNewMetadata()
+        .withName("my-kamelet")
+        .endMetadata()
+        .withNewSpec()
+        .withDefinition(new JSONSchemaPropsBuilder()
+            // blah blah
+            .build())
+        .endSpec()
+        .build();
 
     assertNotNull(kamelet);
     assertEquals("my-kamelet", kamelet.getMetadata().getName());
@@ -54,8 +54,8 @@ class KameletModelTest {
     final ObjectMapper mapper = new ObjectMapper();
     // Given
     String originalJson = new Scanner(getClass().getResourceAsStream("/valid-kamelet.json"))
-      .useDelimiter("\\A")
-      .next();
+        .useDelimiter("\\A")
+        .next();
 
     // When
     final Kamelet kamelet = mapper.readValue(originalJson, Kamelet.class);
@@ -72,5 +72,29 @@ class KameletModelTest {
     assertTrue(kamelet.getSpec().getTemplate().containsKey("beans"));
     assertTrue(kamelet.getSpec().getTemplate().containsKey("from"));
   }
-}
 
+  @Test
+  void shouldDeserializeValidYamlContainingCamelJSONSchemaProps() throws JsonProcessingException {
+    final ObjectMapper mapper = new ObjectMapper();
+    // Given
+    String originalJson = new Scanner(getClass().getResourceAsStream("/valid-kamelet-x-descriptors.json"))
+        .useDelimiter("\\A")
+        .next();
+
+    // When
+    final Kamelet kamelet = mapper.readValue(originalJson, Kamelet.class);
+    final String serializedJson = mapper.writeValueAsString(kamelet);
+    final Kamelet kameletFromSerializedJson = mapper.readValue(serializedJson, Kamelet.class);
+
+    // Then
+    assertNotNull(kamelet);
+    assertNotNull(serializedJson);
+    assertNotNull(kameletFromSerializedJson);
+    assertEquals(kamelet.getMetadata().getName(), kameletFromSerializedJson.getMetadata().getName());
+    assertNotNull(kamelet.getSpec().getDefinition());
+    assertNotNull(kamelet.getSpec().getDefinition().getProperties());
+    assertNotNull(kamelet.getSpec().getDefinition().getProperties().get("validate"));
+    assertEquals(Collections.singletonList("urn:alm:descriptor:com.tectonic.ui:checkbox"),
+        kamelet.getSpec().getDefinition().getProperties().get("validate").getxDescriptors());
+  }
+}

--- a/extensions/camel-k/model-v1alpha1/src/test/resources/valid-kamelet-x-descriptors.json
+++ b/extensions/camel-k/model-v1alpha1/src/test/resources/valid-kamelet-x-descriptors.json
@@ -1,0 +1,92 @@
+{
+  "apiVersion": "camel.apache.org/v1alpha1",
+  "kind": "Kamelet",
+  "metadata": {
+    "name": "foo-deserialize-action",
+    "annotations": {
+      "camel.apache.org/kamelet.support.level": "Preview",
+      "camel.apache.org/catalog.version": "main-SNAPSHOT",
+      "camel.apache.org/provider": "Apache Software Foundation",
+      "camel.apache.org/kamelet.group": "Actions"
+    },
+    "labels": {
+      "camel.apache.org/kamelet.type": "action"
+    }
+  },
+  "spec": {
+    "definition": {
+      "title": "Foo Deserialize Action",
+      "description": "Deserialize payload to Foo",
+      "type": "object",
+      "required": [
+        "schema"
+      ],
+      "properties": {
+        "schema": {
+          "title": "Schema",
+          "description": "The Foo schema to use during serialization (as single-line, using JSON format)",
+          "type": "string",
+          "example": "{\"type\": \"record\", \"namespace\": \"com.example\", \"name\": \"FullName\", \"fields\": [{\"name\": \"first\", \"type\": \"string\"},{\"name\": \"last\", \"type\": \"string\"}]}"
+        },
+        "validate": {
+          "title": "Validate",
+          "description": "Indicates if the content must be validated against the schema",
+          "type": "boolean",
+          "default": true,
+          "x-descriptors": [
+            "urn:alm:descriptor:com.tectonic.ui:checkbox"
+          ]
+        }
+      }
+    },
+    "dependencies": [
+      "github:apache.camel-kamelets:camel-kamelets-utils:main-SNAPSHOT",
+      "camel:kamelet",
+      "camel:core",
+      "camel:jackson-foo"
+    ],
+    "template": {
+      "from": {
+        "uri": "kamelet:source",
+        "steps": [
+          {
+            "set-property": {
+              "name": "schema",
+              "constant": "{{schema}}"
+            }
+          },
+          {
+            "set-property": {
+              "name": "validate",
+              "constant": "{{validate}}"
+            }
+          },
+          {
+            "unmarshal": {
+              "foo": {
+                "library": "Jackson",
+                "unmarshalType": "com.fasterxml.jackson.databind.JsonNode",
+                "schemaResolver": "#class:org.apache.camel.kamelets.utils.serialization.InflightFooSchemaResolver"
+              }
+            }
+          },
+          {
+            "remove-property": {
+              "name": "schema"
+            }
+          },
+          {
+            "remove-property": {
+              "name": "validate"
+            }
+          },
+          {
+            "remove-header": {
+              "name": "Content-Type"
+            }
+          }
+        ]
+      }
+    }
+  }
+}

